docs/guides/javascript/index.md

```diff
@@ -495,6 +495,35 @@ Prefetch.prefetchTemplate('core/toast');
 Prefetch.prefetchTemplates(['core/toast', 'core/modal']);
 ```
 
+## Dropzone
+
+<Since versions={[ 4.4 ]} />
+
+The use of `core/dropzone` modules provides a simplified developer experience for creating dropzones within Moodle.
+
+The module attempts to ensure that accessibility requirements are met, including applying the correct styles and keyboard navigation.
+
+Dropzones will fire trigger callbacks for common actions that occur within the dropzone for other code to listen to and react accordingly.
+
+```js title="Example of creating a dropzone"
+import Dropzone from 'core/dropzone';
+
+// Get the element that will be the dropzone.
+const dropZoneEle = document.querySelector('#dropZoneId');
+// Create a new dropzone accepting only images.
+const dropZone = new Dropzone(
+    dropZoneEle,
+    'image/*',
+    function(files) {
+        window.console.log(files);
+    }
+);
+// Set the custom message for the dropzone. Otherwise, it will use the default message.
+dropZone.setLabel('Drop images here');
```

Comment:
Just noting that it's a shame that this method does not accept a Promise.

Reply:
Yep, I will create a follow-up one to make it support.